swfiles/sw_fitpowder.m

```diff
@@ -428,7 +430,7 @@ function clear_cache(obj)
                 resid = resid./(obj.e);
             end
             % exclude nans in both ycalc and input data
-            ikeep = isfinite(resid);
+            ikeep = isfinite(resid) & obj.e > 1e-12;
```

Comment:
Where does 1e-12 come from? Why not check if obj.e is finite?

Reply:
Thanks for picking this up, sorry I meant to label it as draft as  the scope has widened and I was going to do some more changes after feedback from the SpinW workshop.  I also need to add tests...

But to answer your question, it seems like bins outside the detector coverage are `NaN` - though I wasn't sure this was always the case. Additionally there could be bins where there is coverage but low flux such that you get bins with 0 counts and 0 error. It seems like the data are normalised (not in raw counts) so the 0 is a float. So I want to exclude 0 (or within machine precision of 0) bins but I could in principle have errors that are near zero but valid (depending on the normalisation). I haven't decided the best way to do it yet so I just picked an arbitrary small number and thought I'd figure it out later!

Reply:
Probably `1e-12` is too high and I can just use `eps` - but if any empty bins do slip through they can have a very large impact on the cost function!